pkg/gen/typegen.go

```diff
@@ -418,6 +419,42 @@ func nodeJSProvider() groupOpts { return groupOpts{generatorType: provider, lang
 
 func pythonProvider() groupOpts { return groupOpts{generatorType: provider, language: python} }
 
+func allCamelCasePropertyNames(definitionsJSON map[string]interface{}, opts groupOpts) []string {
+	// Map definition JSON object -> `definition` with metadata.
+	definitions := make([]*definition, 0)
+	linq.From(definitionsJSON).
+		WhereT(func(kv linq.KeyValue) bool {
+			// Skip these objects, special case. They're deprecated and empty.
+			defName := kv.Key.(string)
+			return !strings.HasPrefix(defName, "io.k8s.kubernetes.pkg")
+		}).
+		SelectT(func(kv linq.KeyValue) *definition {
+			defName := kv.Key.(string)
+			return &definition{
+				gvk:  gvkFromRef(defName),
+				name: defName,
+				data: definitionsJSON[defName].(map[string]interface{}),
+			}
+		}).
+		ToSlice(&definitions)
+
+	properties := sets.String{}
+	// Only select camel-cased property names
+	re := regexp.MustCompile(`[a-z]+[A-Z]`)
```

Comment:
My only reservation is this regex. I personally would have just checked whether there is both a lower case and an upper case letter in the string, and I would use the Go string functions to do it, since they (as a minor point) cope with unicode.

Reply:
I decided to just stick with the regex since property names are unlikely to use unicode, and the logic is a bit simpler.